packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -164,17 +162,25 @@ const Tooltip: React.FC<TooltipProps> = (props: ScopedProps<TooltipProps>) => {
         // we dispatch here so `TooltipProvider` isn't required to
         // ensure other tooltips are aware of this one opening.
         document.dispatchEvent(new CustomEvent(TOOLTIP_OPEN));
-      } else {
-        isFocusOpenRef.current = false;
+        onOpen();
       }
-      window.clearTimeout(openTimerRef.current);
       onOpenChange?.(open);
     },
   });
-
   const stateAttribute = React.useMemo(() => {
-    return open ? (wasOpenDelayed ? 'delayed-open' : 'instant-open') : 'closed';
-  }, [wasOpenDelayed, open]);
+    return open ? (wasOpenDelayedRef.current ? 'delayed-open' : 'instant-open') : 'closed';
+  }, [open]);
+
+  const handleOpen = React.useCallback(() => {
```

Comment:
Could this be name `handleOpenInstantly` or something like that?
It's all getting a bit hard to follow now in `onTriggerEnter` with the branching + the external handler.

Reply:
I feel like `handleOpen` and `handleClose` is already instant... otherwise should we rename all of our `handleOpen` handlers in our other comps?  It already implies that it's instant. 

If it helps, I've abstracted the delayed logic into a `handleDelayedOpen` so that you can see easily that sometimes it is delayed.

Reply:
Yeah that works for me.